examples/embeddable_examples/public/react_embeddables/field_list/types.ts

```diff
@@ -19,7 +19,12 @@ export type FieldListSerializedStateState = SerializedTitles & {
   selectedFieldNames?: string[];
 };
 
-export type FieldListApi = DefaultEmbeddableApi & PublishesSelectedFields & PublishesDataViews;
+export type FieldListApi = DefaultEmbeddableApi<
+  FieldListSerializedStateState,
+  FieldListSerializedStateState
```

Comment:
Nit: The RuntimeState type matches the SerializedState type, which makes sense to me. But I wonder if, __for the examples__, we should specifically add a RuntimeState type like `type FieldListRuntimeState = FieldListSerializedStateState` to make it clearer for engineers jumping into the new framework?

Or maybe just a comment explaining that SerializedState and RuntimeState are equivalent?

Reply:
we could also add RuntimeState and SerializedState to embeddable plugin readme.md to help explain why the 2 versions of state exist and an example (writen, not code) of when they should be different

Reply:
@nickpeihl I think that is a good idea. I've seen that pattern used in quite a few of the embeddables made by other teams too.

I only wish there was a way to default it to the serialized state and allow implementors to skip adding it - but that isn't possible because of the order of the template. 

@nreese I think the documentation is also a good idea, but if it's okay I'll skip that for this PR so that we can get it in ASAP to unblock https://github.com/elastic/kibana/pull/186052

Reply:
Added this in ab4326ee955ee8f6c76de71ad55db6af689b15ec